### ISSUE TITLE
[Firestore] Invalid query limit on iOS only

### ISSUE THREAD

**Issue by jokopurnomoa** [invertase/react-native-firebase#586]
### Issue

Hi,
I want to make a chat using firestore, but when using query limit I have an error on iOS. This query runs nicely on android.

![firestore query limit error](https://user-images.githubusercontent.com/5596662/32422205-8799d464-c2d1-11e7-8f55-39f1e7e3d557.png)

```
const _this = this;
this.unsibscribeChat = this.chatRef
    .orderBy('createdAt', 'desc')
    .limit(10)
    .onSnapshot(function(querySnapshot) {
                _this.messageList = [];

                querySnapshot.forEach(function(doc) {
                    // console.log(doc.id, " => ", doc.data());

                    const message = doc.data();
                    message.id = doc.id;
                    _this.messageList.push(message);
                });
            });
```
### Environment

1. Application Target Platform: iOS

2. Development Operating System: Mac OS X El Capitan

3. Build Tools: iOS 10.2 Simulator

4. React Native version: 0.49.3

5. RNFirebase Version: 3.0.6

6. Firebase Module: Cloud Firestore

**Comment by chrisbianca** [comment]
@jokopurnomoa This should be fixed in v3.1.0 which will be released later today.  I'll update this issue when it has been.

**Comment by Salakar** [comment]
@jokopurnomoa v3.1.0 is out now with this fix. Thanks for the report
